core/src/main/java/edu/wpi/grip/core/Pipeline.java

```diff
@@ -69,6 +68,15 @@ public void clear() {
         return Collections.unmodifiableSet(this.connections);
     }
 
+    /*
+     * @return The current per-project settings.  This object may become out of date if the settings are edited
+     * by the user, so objects requiring a preference value should also subscribe to {@link ProjectSettingsChangedEvent}
+     * to get updates.
+     */
+    public ProjectSettings getProjectSettings() {
+        return settings;
```

Comment:
I don't think it's worth it to do that.  That would just add another thing we have to update when we add more settings, and in return it adds almost no value.